spec/models/mixins/deprecation_mixin_spec.rb

```diff
@@ -1,16 +1,4 @@
 RSpec.describe DeprecationMixin do
-  # Host.deprecate_attribute :address, :hostname
-  context ".arel_table" do
-    # this is defining an alias
-    # it is not typical for aliases to work through arel_table
-    # may need to get rid of this in the future
-    it "works for deprecate_attribute columns" do
-      expect(Host.attribute_supported_by_sql?(:address)).to eq(true)
```

Comment:
This are_table context was testing the internals. We're no longer using virtual columns or alias_attribute as someone could use deprecate_attribute on associations or attributes.  The former won't work in rails 7.1 without warnings and will not work with virtual columns.  It's much simpler to handle attributes or associations through runtime delegations although there is some performance implications we're accepting for this flexibility.  We can bring that back in a different way if we need it.